src/System.Net.Security/src/System/Net/CertificateValidationPal.Windows.cs

```diff
@@ -187,12 +186,12 @@ internal static X509Store EnsureStoreOpened(bool isMachineStore)
         {
             X509Store store = isMachineStore ? s_myMachineCertStoreEx : s_myCertStoreEx;
 
-            if (store == null)
+            if (Volatile.Read(ref store) == null)
             {
                 lock (s_syncObject)
                 {
                     store = isMachineStore ? s_myMachineCertStoreEx : s_myCertStoreEx;
-                    if (store == null)
+                    if (Volatile.Read(ref store) == null)
```

Comment:
IIRC the second volatile read is not necessary--the lock above provides the necessary fences.